app/controllers/hxlstats_controller.rb

```diff
@@ -0,0 +1,69 @@
+class HxlstatsController < ApplicationController
+  # Class to collect and aggregate data to parse to the HXL proxy/endpoint
+  def new
+  end
+
+  def create
+  end
+
+  # rubocop:disable MethodLength
+  def show
+    # Prepare output results array
```

Comment:
Thanks for working all this grouping logic out! Tested locally and so exciting to finally have an endpoint with data.
 👏 

A couple of suggestions for cleaning it up a little bit, keeping Rubocop happy and also making testing a bit easier:

1) Move everything from line 12 to 59, i.e. creating the `results` array, out of the controller. One suggestion is making it a public class method in the `HxlStatsView` model, like `results_by_emotional_state`. Or another option is to create a new service object / Plain Old Ruby Object (PORO) like `drawings_by_emotional_state_query` in a new directory like `app/models/queries`, to save from bloating the model also.

2) Once the logic is extracted, you can extract each block below into its own private method, which will be easier to read / maintain / identify the purpose of each block by explicit method naming - this last bit also makes it less necessary to feel like you need to prefix comments on each block, as the methods should tell the tale. AND this will make you fight with Rubocop less and remove the need for `# rubocop:disable MethodLength` I see stuck up there :)

3) Once the logic is extracted outside of the controller, testing becomes a bit clearer/easier. For controller specs (not detrimental if these are missing btw), you can set up just a couple of tests that focus on asserting the _response_ of the `show` method (i.e. returns html if normal request, returns data if json specified) and just stub the value of the results generation method. Then you can focus on testing the actual results generation in the model / service object, using FactoryGirl for HxlStatsView.

As always, buzz or let me know if you'd like to go through this or pair!